### DISCUSSION TITLE
Only creates a Classic Load balancer

### DISCUSSION THREAD

**Discussion by sethlfuller** [argoproj/argo-helm#2507] [General]
Does the argo-cd chart support AWS Application Load balancers? Even when I set annotations like the one below, I am only getting the creation of Classic Load balancers in AWS via the AWS Load Balancer Controller from https://aws.github.io/eks-charts. Does anyone know what I might be doing wrong or be able to provide the correct settings if ALB creation is supported? Thanks in advance for any advice.

```
 set {
     name = "server.ingress.annotations.kubernetes\\.io/ingress\\.class"
     value = "alb"
      }
```

Or even:

```
server:
  ingress: 
    enabled: true
    className: "alb"
```

**Comment by pdrastil** [comment]
Hi @sethlfuller - yet it does. See example in docs:
https://github.com/argoproj/argo-helm/tree/main/charts/argo-cd#aws-application-load-balancer

**Reply by sethlfuller** [reply]
This is great thank you so much. I made some progress but still no luck. I put this in my helm_release resource also:

```
 set {
    name = "server.service.type"
    value = "LoadBalancer"
  }
```

I see that it's now creating an extra service "service/argocd-server-grpc" but it remains as a ClusterIP service type. Should this be a LoadBalancer service type? With my current configuration, the service/argocd-server  service is what is being setup with the Load Balancer, and it is a Classic Load Balancer.

Also is there anything else I need to set statically in the helm_release resource? The only other things I'm setting are the cluster name and the hostname.

**Reply by pdrastil** [reply]
I think you mixing things a little. AWS ALB controller allows you to provision 2 different types of load balancers in AWS. 

1) Network Load Balancers (NLB) that are routing on layer 4 and maps to K8s Services with type LoadBalancer
2) Application Load Balancers (ALB) that are routing on layer 7 and maps to K8s Ingress resources. 

See: https://kubernetes-sigs.github.io/aws-load-balancer-controller

If you plan to use ALB setup with Ingress resource and `ingressClass: alb` the aws-load-balancer-controller will act as ingress controller for your cluster. Using `ClusterIP` or `NodePort` for depends if you want AWS target group route to all EC2 instances (instance mode - NodePort) or directly to Pods (IP mode - ClusterIP).

If you plan to use NLB, you can either create service type LoadBalancer without ingress to get dedicated load balancer just for Argo CD server (expensive option) or deploy different ingress controller that will use NLB and provide another `ingressClass` implementation where ingress guide above will not apply anymore.

The example I've sent is specific only for AWS ALB mode. As you've noticed there is additional service called `argocd-server-grpc` that is used for routing gRPC traffic to server and already contains necessary AWS annotations. See operator manual for more details: https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#aws-application-load-balancers-albs-and-classic-elb-http-mode

**Reply by sethlfuller** [reply]
Thank you so much. I figured out that my AWS Load Balancer Controller IAM policy had the wrong permissions. Once I corrected that, the ALB was created without any further configuration of my Argo CD Helm install. I read somewhere that EKS can create CLB's without any special permissions, so that's what clued me into the possibility that my LB Controller wasn't working correctly.
